egs/swbd/run.sh

```diff
@@ -57,6 +57,18 @@ for order in 3 4 5; do
     format_arpa_lm.py data/lm_20k_${order}_prune${threshold} | gzip -c > data/arpa/20k_${order}gram_prune${threshold}.arpa.gz
 
   done
+
+# example of pruning by size.
```

Comment:
I tried to run examples (swbd and swbd_fisher), but it seems to have some bug like this

+ for threshold in 1.0 0.25
+ prune_lm_dir.py data/lm_20k_3 1.0 data/lm_20k_3_prune1.0
+ tail -n 5
+ head -n 3
*** glibc detected *** merge-float-counts: double free or corruption (!prev): 0x00000000018570c0 ***
======= Backtrace: =========
/lib/x86_64-linux-gnu/libc.so.6(+0x75bb6)[0x7f7f17a79bb6]
/lib/x86_64-linux-gnu/libc.so.6(cfree+0x6c)[0x7f7f17a7e95c]
merge-float-counts[0x405fdc]
merge-float-counts[0x4045c2]
merge-float-counts[0x4053f1]
merge-float-counts[0x403c6a]
merge-float-counts[0x402d9f]
merge-float-counts[0x40223b]
merge-float-counts[0x40189a]
merge-float-counts[0x40173a]
/lib/x86_64-linux-gnu/libc.so.6(__libc_start_main+0xfd)[0x7f7f17a22ead]
merge-float-counts[0x4015a9]
======= Memory map: ========
00400000-00422000 r-xp 00000000 00:24 11118193                           /export/a09/zhouyang/pocolm/src/merge-float-counts
00622000-00623000 rw-p 00022000 00:24 11118193                           /export/a09/zhouyang/pocolm/src/merge-float-counts
01852000-01873000 rw-p 00000000 00:00 0                                  [heap]
7f7f10000000-7f7f10021000 rw-p 00000000 00:00 0 
7f7f10021000-7f7f14000000 ---p 00000000 00:00 0 
7f7f17a04000-7f7f17b88000 r-xp 00000000 fe:00 75735                      /lib/x86_64-linux-gnu/libc-2.13.so
7f7f17b88000-7f7f17d87000 ---p 00184000 fe:00 75735                      /lib/x86_64-linux-gnu/libc-2.13.so
7f7f17d87000-7f7f17d8b000 r--p 00183000 fe:00 75735                      /lib/x86_64-linux-gnu/libc-2.13.so
7f7f17d8b000-7f7f17d8c000 rw-p 00187000 fe:00 75735                      /lib/x86_64-linux-gnu/libc-2.13.so
7f7f17d8c000-7f7f17d91000 rw-p 00000000 00:00 0 
7f7f17d91000-7f7f17da6000 r-xp 00000000 fe:00 49060                      /lib/x86_64-linux-gnu/libgcc_s.so.1
7f7f17da6000-7f7f17fa6000 ---p 00015000 fe:00 49060                      /lib/x86_64-linux-gnu/libgcc_s.so.1
7f7f17fa6000-7f7f17fa7000 rw-p 00015000 fe:00 49060                      /lib/x86_64-linux-gnu/libgcc_s.so.1
7f7f17fa7000-7f7f18028000 r-xp 00000000 fe:00 75739                      /lib/x86_64-linux-gnu/libm-2.13.so
7f7f18028000-7f7f18227000 ---p 00081000 fe:00 75739                      /lib/x86_64-linux-gnu/libm-2.13.so
7f7f18227000-7f7f18228000 r--p 00080000 fe:00 75739                      /lib/x86_64-linux-gnu/libm-2.13.so
7f7f18228000-7f7f18229000 rw-p 00081000 fe:00 75739                      /lib/x86_64-linux-gnu/libm-2.13.so
7f7f18229000-7f7f18311000 r-xp 00000000 fe:00 1629156                    /usr/lib/x86_64-linux-gnu/libstdc++.so.6.0.17
7f7f18311000-7f7f18511000 ---p 000e8000 fe:00 1629156                    /usr/lib/x86_64-linux-gnu/libstdc++.so.6.0.17
7f7f18511000-7f7f18519000 r--p 000e8000 fe:00 1629156                    /usr/lib/x86_64-linux-gnu/libstdc++.so.6.0.17
7f7f18519000-7f7f1851b000 rw-p 000f0000 fe:00 1629156                    /usr/lib/x86_64-linux-gnu/libstdc++.so.6.0.17
7f7f1851b000-7f7f18530000 rw-p 00000000 00:00 0 
7f7f18530000-7f7f18550000 r-xp 00000000 fe:00 75732                      /lib/x86_64-linux-gnu/ld-2.13.so
7f7f18665000-7f7f18732000 rw-p 00000000 00:00 0 
7f7f1874b000-7f7f1874f000 rw-p 00000000 00:00 0 
7f7f1874f000-7f7f18750000 r--p 0001f000 fe:00 75732                      /lib/x86_64-linux-gnu/ld-2.13.so
7f7f18750000-7f7f18751000 rw-p 00020000 fe:00 75732                      /lib/x86_64-linux-gnu/ld-2.13.so
7f7f18751000-7f7f18752000 rw-p 00000000 00:00 0 
7ffcbac6a000-7ffcbac8b000 rw-p 00000000 00:00 0                          [stack]
7ffcbada9000-7ffcbadaa000 r-xp 00000000 00:00 0                          [vdso]
ffffffffff600000-ffffffffff601000 r-xp 00000000 00:00 0                  [vsyscall]

On Jul 27, 2016, at 5:14 PM, Daniel Povey <notifications@github.com> wrote:

> In egs/swbd/run.sh:
> 
> > @@ -57,6 +57,18 @@ for order in 3 4 5; do
> >      format_arpa_lm.py data/lm_20k_${order}_prune${threshold} | gzip -c > data/arpa/20k_${order}gram_prune${threshold}.arpa.gz
> >  
> >    done
> > +
> > +# example of pruning by size.
> I'm a little uncomfortable with having the example script pruning to a proportion like this, because people might think it's a pattern to follow in real-life examples. In real life ASR applications you'd probably want to prune to a fixed target, like 1.5 million n-grams. 
> BTW, make sure your LM-pruning script dies if the target number of n-grams is >= the initial number of n-grams in the LM, or is already within the threshold [in which case there is nothing to do, so best to skip the pruning stage.]
> 
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub, or mute the thread.
>